docs/support/tutorials/ml-llm.md

```diff
@@ -180,33 +180,101 @@ Face](https://huggingface.co/blog/mlabonne/sft-llama3), which also
+
+It can take hours to quantize very large models, but luckily many models already have a quantized version available, for example in Hugging Face. You can look for quantized models by a suffix in the model name indicating a quantization method, such as AWQ, GPTQ, or GGUF, or alternatively, model precision, such as 8bit or 4bit. 
+
+### Using BitsandBytes Quantization
+
+The [BitsAndBytes library from Hugging Face Transformers](https://huggingface.co/docs/transformers/en/quantization/bitsandbytes) offers runtime quantization, which compresses weights to 8-bit or 4-bit on-the-fly during inference to save memory.
 
 ```python
-from transformers import BitsAndBytesConfig
+from transformers import BitsAndBytesConfig, AutoModelForCausalLM
+
 bnb_config = BitsAndBytesConfig(
-    load_in_4bit=True,
-    bnb_4bit_quant_type="nf4",
-    bnb_4bit_compute_dtype=torch.bfloat16,
-    bnb_4bit_use_double_quant=True,
-    bnb_4bit_quant_storage=torch.bfloat16,
+   load_in_4bit=True,
+   bnb_4bit_quant_type="nf4",
+   bnb_4bit_compute_dtype=torch.bfloat16,
+   bnb_4bit_use_double_quant=True,
+   bnb_4bit_quant_storage=torch.bfloat16,
+)
+
+model = AutoModelForCausalLM.from_pretrained(
+   args.model,
+   quantization_config=bnb_config,
+   ...
+)
+```
+You can use it in our fine-tuning LLMs example (see section above) with the --4bit argument. Alternatively, see our [Github repository](https://github.com/CSCfi/llm-quantization-scripts/tree/main/BitsAndBytes) for a full example to quantize a model using bitsandbytes on Puhti, Mahti or LUMI.
+
+### Using GPTQ Quantization
+
+In addition to bitsandbytes, Hugging Face Transformers has also integrated [Gradient Post-Training Quantization (GPTQ)](https://huggingface.co/docs/transformers/en/quantization/gptq). 
+[GPTQ](https://arxiv.org/abs/2210.17323) performs row-wise quantization of weight matrices, optimizing each row individually so that the quantized weights closely approximate the original values with minimal reconstruction error.  
+
+Unlike runtime quantization libraries such as bitsandbytes, GPTQ performs a one-time compression step using a calibration dataset, resulting in a new quantized model that can be saved and loaded without requiring the original full-precision weights. To use GPTQ quantization from transformers, create a [GPTQConfig](https://huggingface.co/docs/transformers/v4.56.2/en/main_classes/quantization#transformers.GPTQConfig) class with number of bits to quantize to, a dataset to calibrate the weights for quantization, and a tokenizer.
+
+```python
+from transformers import GPTQConfig, AutoModelForCausalLM
+
+gptq_config = GPTQConfig(
+    bits=4,            
+    dataset="c4",        
+   tokenizer=tokenizer   
```

Comment:
tokenizer is indented one space less than the other lines - Python will be strict about this :)